### DISCUSSION TITLE
Building and using static libraries on windows.

### DISCUSSION THREAD

**Discussion by dmhursh** [ImageMagick/ImageMagick#7372] [Help] [answered]
### ImageMagick version

7.1.1-29-284-gbe06e6c45

### Operating system, version and so on

windows 10 / vs2019

### Description

I'm trying to build static libraries on windows using ImageMagick/ImageMagick-Windows 20200615-128-g7a45f83.

After disabling EMF & EXR I got it to build.  Now I'm trying to use what I've built.  I think I've found the .lib files, build I can't find `ImageMagick.h` or the `magick` directory anywhere causing the include do fail.

``` C
#include <magick/ImageMagick.h>
```

Based on https://github.com/ImageMagick/ImageMagick/discussions/5645 I assume I'm going to need to figure out more compile and link options once I get past this.  I really wish the pkgconfig stuff worked on all the platforms.

I assume I'm doing something stupid.  Any help would be greatly appreciated.

**Comment by dlemstra** [comment, marked as answer]
Why do you think you need to include that file? The `magick` folder is from ImageMagick 6. You should use something like this:
```c
#include <MagickCore/MagickCore.h>
#include <MagickCore/studio.h>
```

**Comment by dmhursh** [comment]
That would be something stupid on my part.  I'm follow example code from http://astroa.physics.metu.edu.tr/MANUALS/ImageMagick-6.2.5/www/magick-core.html.
I see now that it's from 6.2.5.  I'll see if I can find a more up-to-date example.  Thanks.

**Comment by dmhursh** [comment]
Are these examples I should be looking at?
https://imagemagick.org/MagickWand/

**Comment by snibgo** [comment]
> Are these examples I should be looking at?
https://imagemagick.org/MagickWand/

Those examples have not been updated for v7. They use directories `magick` and `wand`, when they are now called `MagickCore` and `MagickWand`.

**Comment by dmhursh** [comment]
It looks like the only available example now is https://github.com/ImageMagick/ImageMagick/blob/main/api_examples/wand.c

I have it compiling.  Now I just need to resolve the link errors.  Thanks

**Comment by dmhursh** [comment]
I got it to work with the following.  The autoconf replacement stuff for windows needs more documentation.

``` cmake
ADD_EXECUTABLE(TESTTOOL testtool.c)
	# Match the quantum and HDRI from Configure.exe
	# The others make the static link work.
	TARGET_COMPILE_DEFINITIONS(TESTTOOL PRIVATE
		MAGICKCORE_QUANTUM_DEPTH=32
		MAGICKCORE_HDRI_ENABLE=1
		_VISUALC_
		_MAGICKLIB_
		)
 
	FILE(GLOB IMAGEMAGICK_LIBS "deps/lib/CORE_RL_*.lib")
	TARGET_INCLUDE_DIRECTORIES(TESTTOOL PRIVATE "${MY_IM_PATH}/git/ImageMagick-Windows/ImageMagick")
	TARGET_LINK_DIRECTORIES(TESTTOOL PRIVATE "deps/lib")
	TARGET_LINK_LIBRARIES(TESTTOOL PRIVATE "${IMAGEMAGICK_LIBS}")
